### DISCUSSION TITLE
how to use MITM on Windows only for external devices (not local device connections)?

### DISCUSSION THREAD

**Discussion by aledelgo** [mitmproxy/mitmproxy#6611] [General]
I've installed mitm on Windows 11.
local ethernet connection set with PROXY: OFF
when i open any browser, all connection go trough mitm (infact i see SSL error on Chrome).

i need instead to use MITM only for external devices (phone, virtual machines and other PC in my LAN) setting their gateway address to my Win11PC+MITM.

how mitm autoconfigure windows to capture all local traffic without asking a permission? how to exclude this behaviour?

thanks

**Comment by mhils** [comment]
mitmproxy does not automatically configure anything to go through mitmproxy (at least in regular proxy mode), not sure what you are experiencing here.

**Comment by aledelgo** [comment]
same here, can't believe it's happening but it's happening.
when i start from Terminal "mitmweb --mode transparent --showhost" any browser on the same host can't navigate anymore (tried chrome and firefox) as any website give SSL certificate error and when i see details i see the certificate is issude by mitm.

checked if something in the routing changed but nexthop is anyway the external gateway.

i can see in the terminal log a lot of connections but i don't know why, in the attached screen just after launching mitmweb i tried to open gmail.com (other browser tab was already open).

![image](https://github.com/mitmproxy/mitmproxy/assets/1503438/ae501524-b68f-46e4-aad7-81caa06c50a5)

**Reply by mhils** [reply]
Transparent mode on Windows is/was never officially supported. https://github.com/mitmproxy/mitmproxy/blob/10.2.2/mitmproxy/platform/windows.py has some notes on what's up here, but I'd very strongly recommend 1) regular mode or 2) WireGuard mode instead.

**Reply by aledelgo** [reply]
the line 365 tells just what i'm experiencing:

    Transparent Windows Proxy for mitmproxy based on WinDivert/PyDivert. This module can be used to
    **redirect** both traffic that is forwarded by the host and **traffic originating from the host itself**.

so it is expected...

**Reply by mhils** [reply]
You can spawn the redirector manually before starting mitmproxy, see further down in the same file. But again, transparent mode on Windows remains formally unsupported for now.

**Reply by aledelgo** [reply]
i see in the code the self routing mode is enabled by a param "local" in the constructor... is there a way to set this to false trough command line parameters?
